### ISSUE TITLE
Implement FlutureTMonetEither.when/.whenF operator

### ISSUE THREAD

**Issue by char0n** [char0n/monadT#6]
Example
```javascript
FlutureTMonetEither
  .fromValue(1)
  .when(equals(1), add(1))
```
```javascript
FlutureTMonetEither
  .fromValue(1)
  .whenF(equals(1), v => FlutureTMonetEither.fromValue(v + 1))
```
`when` does conditional mapping under the hood
`whenF` does conditional chaining under the hood

@michaelkuk does this make sense ?

**Comment by michaelkuk** [comment]
@charOn It definitely does, however I am not sure whether the `when` operator is needed as this functionality can easily be supplemented by using map operator with `Ramda.when` function. On the other hand it would be nice not having to import extra dependency for this usecase.

The `whenF` operator is most definitely feature worth adding.

**Comment by char0n** [comment]
@michaelkuk yes, you can easily achieve `when` operator with `map(when(...))`, but the point of creating custom operators is to create abstraction for most common usecases. And I personally see this as common usecase as already existing `tap` operator.

**Comment by michaelkuk** [comment]
@char0n true enough, I shall look at the implementation.
